src/python/grpcio_reflection/grpc_reflection/v1alpha/reflection.py

```diff
@@ -133,13 +54,47 @@ def ServerReflectionInfo(self, request_iterator, context):
                     ))
 
 
-def enable_server_reflection(service_names, server, pool=None):
-    """Enables server reflection on a server.
+if sys.version_info[0] >= 3 and sys.version_info[1] >= 6:
+    # Exposes AsyncReflectionServicer as public API.
+    from . import _async as aio
+    from grpc.experimental import aio as grpc_aio
 
-    Args:
-      service_names: Iterable of fully-qualified service names available.
-      server: grpc.Server to which reflection service will be added.
-      pool: DescriptorPool object to use (descriptor_pool.Default() if None).
-    """
-    _reflection_pb2_grpc.add_ServerReflectionServicer_to_server(
-        ReflectionServicer(service_names, pool=pool), server)
+    def enable_server_reflection(service_names, server, pool=None):
+        """Enables server reflection on a server.
+
+        Args:
+            service_names: Iterable of fully-qualified service names available.
+            server: grpc.Server to which reflection service will be added.
+            pool: DescriptorPool object to use (descriptor_pool.Default() if None).
+        """
+        if isinstance(server, grpc_aio.Server):
+            _reflection_pb2_grpc.add_ServerReflectionServicer_to_server(
+                aio.ReflectionServicer(service_names, pool=pool), server)
+        else:
+            _reflection_pb2_grpc.add_ServerReflectionServicer_to_server(
+                ReflectionServicer(service_names, pool=pool), server)
+
+    __all__ = [
+        "SERVICE_NAME",
+        "ReflectionServicer",
+        "enable_server_reflection",
+        "aio",
+    ]
+else:
+
+    def enable_server_reflection(service_names, server, pool=None):
+        """Enables server reflection on a server.
```

Comment:
The docstrings end up duplicated though. I'm not sure that there's a great solution for this, but it's something that we need to keep in mind going forward. What do [the generated docs](https://grpc.github.io/grpc/python/grpc_reflection.html) for this look like?

Reply:
Since this is Python, I created a common string literal and assigned them after the function is declared.